### PR TITLE
Offline all the things

### PR DESCRIPTION
**Pull request by stristr**
NOT OK to merge.

Medium review, but we will probably just merge when ready. Early PR to see if there are glaring Jenkins issues.

Summary of changes:

- [x] Retire successful experiments (`MultiComponentFeatures`, `GroupUngroup`, etc.)
- [x] Implement local PNG/GIF/MP4 bakery, including inkstone and app hooks during publish.
  - [x] Improves original implementation by using a queue for PNG sequence baking and parallel exports, skipping PNG sequences that don't need to be generated in the event of many parallel requests, etc.
- [x] Rewire the authentication and project bootstrapping processes to work out of Envoy (almost—I didn't move Master and MasterGitProject out of Plumbing, although it would be relatively easy to do so), and use the Haiku.app "project object" (a.k.a. `remapProjectObjectToExpectedFormat`, an object we get by parsing `inkstone`'s response) as a single source of truth for all project-related IPC. Now uses local registries per Envoy Handler (e.g. `User` keeps a registry in `~/.haiku/registry.json` that keeps track of identity and preferences, `Project` keeps a registry in `~/.haiku/projects/<username>/registry.json` that keeps track of projects for offline stuff, etc.).
- [x] Defer caudex (GitLab) backing until first publish.
- [x] Move public/private opt-in workflow to first publish.
- [x] Provide free user blockades, security features, and upgrade CTAs at various points where users should be blocked.
  - [x] @taylorpoe needs to swoop in and beautify these. All areas requiring design love are with comments containing the string `#FIXME(@taylor)`.
- [x] Unrelated: fixes Angular packaging to work with the latest `@angular/cli`.
- [x] Unrelated: reworks Lottie initialization to actually mount a Haiku component, ensuring we run migrations on legacy bytecode 
- [ ] Offline features (holding a in a separate branch—I will merge here after fixing edge cases):
  - [ ] New project (offline)
  - [ ] Open existing project (offline)
  - [ ] Delete project (offline)
  - [ ] Periodic license checking

Regressions to look for:

- See task: [Write QA checklist for offline, pro, and related features](https://app.asana.com/0/752360003454336/768633613195122/f) (not done yet)

Completed checkin tasks:

- [x] Did manual testing of interrelated functionality
- [x] Wrote an automated test for existing and modified functionality

### CODE REVIEW

File: packages/@haiku/sdk-inkstone/tsconfig.json [hunk 1]
```diff
@@ -12,7 +13,7 @@
     "rootDir": "src",
     "sourceMap": true,
     "suppressImplicitAnyIndexErrors": true,
-    "target": "ES2015",
+    "target": "ES5",
```
Comment (stristr) [inline packages/@haiku/sdk-inkstone/tsconfig.json:16]: 😢 this is the only way to get `share-page` to build against the latest code. (TODO: upgrade ancient `create-react-app` build stack on `share-page`.)

Comment (roperzh) [reply packages/@haiku/sdk-inkstone/tsconfig.json:16]: I think @jonaias also faced this not so long ago :(

**Comment by stristr** [comment]
Note that as of 58e82e1b0a419ad0820c31dbca02dcffc6221b9a, we shouldn't have any tests directly calling inkstone/creating projects/creating repositories/etc.

This is a very good thing, and should even help plumbing tests pass a little faster.

**Comment by stristr** [comment]
@taylorpoe, I'm going to merge this branch to keep `master` fresh after health checks pass—will send an additional PR for checklist items noted here.

(Just a heads up that you can switch to working off `master` with the UI beautification after merge.)

**Comment by stristr** [comment]
LGTJ, merging.

**Pull request merged**